packages/multi-select-combo-box/src/vaadin-multi-select-combo-box.js

```diff
@@ -654,46 +685,37 @@ class MultiSelectComboBox extends ResizeMixin(InputControlMixin(ThemableMixin(El
     }
 
     // Clear all chips except the overflow
-    const chips = Array.from(this._chips).reverse();
-    const overflow = chips.pop();
-
-    chips.forEach((chip) => {
-      chip.remove();
+    Array.from(this._chips).forEach((chip) => {
+      if (chip !== this.$.overflow) {
+        chip.remove();
+      }
     });
 
     const items = [...this.selectedItems];
 
-    let refNode = overflow.nextElementSibling;
+    // Detect available max-width for chips
+    let maxWidth = this.__getChipsMaxWidth();
 
-    // Use overflow chip to measure min-width
-    const chipMinWidth = this.__getMinWidth(overflow);
-    const inputMinWidth = parseInt(getComputedStyle(this.inputElement).flexBasis);
-    const containerStyle = getComputedStyle(this._inputField);
+    if (items.length > 1) {
+      maxWidth -= this.__getOverflowWidth();
+    }
 
-    // Detect available width for chips
-    let totalWidth =
-      parseInt(containerStyle.width) -
-      parseInt(containerStyle.paddingLeft) -
-      parseInt(containerStyle.paddingRight) -
-      this.$.toggleButton.clientWidth -
-      inputMinWidth;
+    this.$.chips.style.maxWidth = `${maxWidth}px`;
 
-    if (this.clearButtonVisible) {
-      totalWidth -= this.$.clearButton.clientWidth;
-    }
+    // Add chips until max width is exceeded
+    for (let i = items.length - 1, totalWidth = 0, refNode = null; i >= 0; i--) {
+      const chip = this.__createChip(items[i]);
+      this.$.chips.insertBefore(chip, refNode);
+
+      totalWidth += chip.clientWidth + parseInt(getComputedStyle(chip).marginInlineEnd);
```

Comment:
nit: parsing these inline margin values from computed styles feels a bit flaky.

(Would be nice to later refactor the chips to use CSS `gap` for spacing, and maybe also so that the overflow chip would be included in the same container as the other chips? That change would break the calculations that rely on inline margin values.)

Maybe the `<input>` offset width could be checked to get the remaining space while adding chips and then break when it reaches the desired value of `--input-min-width`?

Reply:
> Maybe the `<input>` offset width could be checked to get the remaining space while adding chips

I've tried that initially, but in was causing the input to overlap clear button in some cases.
Now when the min-width is removed from chips, this might be worth taking another look.

Reply:
Here is how it looks like when updating the logic to check the input, same problem as I had before:

<img width="354" alt="Screenshot 2022-05-02 at 17 40 12" src="https://user-images.githubusercontent.com/10589913/166253420-10f8d828-0b77-4a6e-8131-c95c2547ac72.png">

Reply:
This is due to the min-width at `flex: 1 0 var(--input-min-width);`. The input doesn't shrink smaller than that because it has `flex-shrink: 0`. One solution could be to have its flex-shrink set to 1 during `__updateChips()`. Maybe `flex-shrink` could always be 1 for the input?

Reply:
But looks like the latest revision no longer uses the computed inline margins so this is good enough IMO. Refactoring this later is not breaking.